Scenario - Windows and SQL Server 2012/2012R2 ESU Database discovery (#2148)
microsoft/azure_arc: docs/azure_arc_jumpstart/azure_arc_sqlsrv/day2/esu/_index.md

---
type: docs

> title: "Using Azure Arc to deliver Extended Security Updates for Windows Server 2012 and SQL Server 2012"
> linkTitle: "Using Azure Arc to deliver Extended Security Updates for Windows Server and SQL Server 2012"

Reviewer: ```suggestion
title: "Using Azure Arc to deliver Extended Security Updates (ESU) for Windows Server 2012 and SQL Server 2012"
linkTitle: "Using Azure Arc to deliver Extended Security Updates (ESU) for Windows Server and SQL Server 2012"
```

Reviewer: Make this update on the servers scenario as well

weight: 1
description: >
---

## Using Azure Arc to deliver Extended Security Updates for Windows Server and SQL Server 2012

> The following Jumpstart scenario will guide you on how to use Azure Arc to enroll Windows Server and SQL Server 2012/2012 R2 machines in [Extented Security Updates (ESUs)](https://learn.microsoft.com/windows-server/get-started/extended-security-updates-overview). This scenario creates an Azure VM with Hyper-V installed where the Windows Server and/or SQL Server VMs will run and will be onboard as Azure Arc-enabled server and/or Azure Arc-enabled SQL Server respectively. Once these VMs are registered in Azure you will have visibility into their ESU coverage and can enroll them through the Azure portal one month before Windows Server 2012 end of support.
> In this scenario you can choose between working with Windows Server 2012 R2, SQL Server 2012 Standard edition or both.

Reviewer: ```suggestion
The following Jumpstart scenario will guide you on how to use Azure Arc to enroll Windows Server and SQL Server 2012/2012 R2 machines in [Extented Security Updates (ESUs)](https://learn.microsoft.com/windows-server/get-started/extended-security-updates-overview). This scenario creates an Azure VM with Hyper-V installed where the Windows Server and/or SQL Server VMs will run and will be onboard as Azure Arc-enabled server and/or Azure Arc-enabled SQL Server respectively. Once these VMs are registered in Azure you will have visibility into their ESU coverage and can enroll them through the Azure portal one month before Windows Server 2012 end of support.

In this scenario, you will be able to choose between working with Windows Server 2012 R2, SQL Server 2012 Standard Edition, or both.
```

- Create Azure service principal (SP)

    To connect a VM or bare-metal server to Azure Arc, Azure service principal assigned with the "Contributor" role is required. To create it, login to your Azure account run the below command (this can also be done in [Azure Cloud Shell](https://shell.azure.com/)).

    ```shell
    az login
    subscriptionId=$(az account show --query id --output tsv)
    az ad sp create-for-rbac -n "<Unique SP Name>" --role "Contributor" --scopes /subscriptions/$subscriptionId
    ```

    For example:

    ```shell
    az login
    subscriptionId=$(az account show --query id --output tsv)
    az ad sp create-for-rbac -n "JumpstartArc" --role "Contributor" --scopes /subscriptions/$subscriptionId
    ```

    Output should look like this:

    ```json
    {
    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "displayName": "JumpstartArc",
    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
    }
    ```

>     > **NOTE: If you create multiple subsequent role assignments on the same service principal, your client secret (password) will be destroyed and recreated each time. Therefore, make sure you grab the correct password**.

Reviewer: ```suggestion
    > **NOTE: If you create multiple subsequent role assignments on the same service principal, your client secret (password) will be destroyed and recreated each time. Therefore, make sure you grab the correct password.**
```

- User provides the Bicep template parameter values, either via the portal or editing the parameters file. These parameters values are used throughout the deployment.

- User will run the Bicep template at resource group level.

- User logs in to the Azure VM using Azure Bastion or RDP to trigger the Logon script. The script will:
  - Install Hyper-V
  - Create Windows and/or SQL VMs in Hyper-V
  - Onboard the Hyper-V VMs as Arc-enabled resources

## Bicep template deployment

- Clone the Azure Arc Jumpstart repository

    ```shell
    git clone https://github.com/microsoft/azure_arc.git
    ```

- Edit the parameter file to match your environment. You will need to provide:
  - _`spnClientId`_: the AppId of the service principal you created before.
  - _`spnClientSecret`_: the password of the service principal you created before.
  - _`spnTenantId`_: your Azure AD's tenant ID.
  - _`windowsAdminUsername`_: Windows admin username for your Azure VM.
  - _`windowsAdminPassword`_: password for the Windows admin username.
  - _`deployBastion`_: whether or not you'd like to deploy Azure Bastion to access the Azure VM. Values can be "true" or "false"

>   - _`esu`_: this parameter will allow you to control what VMs will be deployed. It can be:

Reviewer: ```suggestion
  - _`esu`_: this parameter will allow you to control what VMs will be deployed. It can be either:
```

    - _`ws`_: to only deploy a Windows Server 2012 R2 VM that will be registered as an Arc-enabled server.
    - _`sql`_: to only deploy a SQL Server 2012 Standard Edition VM that will be registered as an Arc-enabled SQL Server.
    - _`both`_: to deploy both a Windows Server 2012 R2 VM and a SQL Server 2012 Standard Edition VM that will be Arc-enabled.

>     > **NOTE: Make sure to set the _esu_ parameter to _sql_ or _both_ to have an Arc-enabled SQL server deployed**.

Reviewer: ```suggestion
    > **NOTE: Make sure to set the _esu_ parameter to _sql_ or _both_ to have an Arc-enabled SQL Server deployed.**
```

    ![Screenshot of Azure Bastion session 02](./06.png)

    ![Screenshot of Azure Bastion session 03](./07.png)

- At first login, as mentioned in the "Automation Flow" section, a logon script will get executed. This script was created as part of the automated deployment process.

    ![Screenshot of script's output](./08.png)

    ![Screenshot of script's output](./09.png)

    ![Screenshot of script's output](./10.png)

    ![Screenshot of script's output](./11.png)

- Let the script to run its course and **do not close** the Powershell session, this will be done for you once completed.

    > **NOTE: The script run time is ~10-15 min long.**

- Upon successful run, a new Azure Arc-enabled server and/or Azure Arc-enabled SQL server will be added to the resource group.

  ![Screenshot Azure Arc-enabled resources on resource group](./12.png)

- Now that you have successfully onboarded the Arc-enabled resources, you will be able to manage the Extended Security Updates (ESU) from the Azure Portal:

## Extended Security Updates (ESU)

Now that you have Windows Server 2012 R2 and/or SQL Server 2012 Arc-enabled, you are able to manage ESU licenses for these servers.

> - Navigate to the Azure Arc page

Reviewer: ```suggestion
- Navigate to the Azure Arc page.
```

>   ![Screenshot Azure Arc navigate](./13.png)
> 
> - Select Extended Security Updates in the left pane.
> 
>   ![Screenshot Azure Arc Extended](./14.png)
> 
> - Provision Windows Server 2012 and 2012 R2 Extended Security Update licenses from Azure Arc.
> 
>   ![Screenshot ESU Licenses](./15.png)
> 
>   ![Screenshot ESU Licenses](./16.png)
> 
> - Select one or more Arc-enabled servers to link to an Extended Security Update license.
> 
>   ![Screenshot Enable ESU Licenses](./17.png)

Reviewer: ```suggestion
  ![Screenshot of Azure Arc navigate](./13.png)

- Select Extended Security Updates in the left pane.

  ![Screenshot of Azure Arc Extended](./14.png)

- Provision Windows Server 2012 and 2012 R2 Extended Security Update licenses from Azure Arc.

  ![Screenshot of ESU Licenses](./15.png)

  ![Screenshot of ESU Licenses](./16.png)

- Select one or more Arc-enabled servers to link to an Extended Security Update license.

  ![Screenshot of Enable ESU Licenses](./17.png)
```

> **NOTE: Verify that the databases running on your Azure Arc-enabled SQL server are being discovered. If the databases are not shown in the Azure Portal, follow these steps to fix it.**

Reviewer: ```suggestion
**NOTE: Verify that the databases running on your Azure Arc-enabled SQL Server are being discovered. If the databases are not shown in the Azure Portal, follow these steps to fix it.**
```

> - Select the Azure Arc-enabled SQL server.

Reviewer: ```suggestion
- Select the Azure Arc-enabled SQL Server.
```

>  ![Screenshot Azure Arc-enabled SQL Server resource](./18.png)
> 
> - Select "Databases" under "Data Management". Click on "Change License Type".
> 
>  ![Screenshot Azure Arc-enabled SQL Server resource](./19.png)
> 
>  ![Screenshot Change License Type](./20.png)
> 
> - Choose "License with Software Assurance".
> 
>  ![Screenshot software assurance](./21.png)
> 
> - The databases will be discovered and shown in the Azure Portal after a few seconds.
> 
>  ![Screenshot Databases in Portal](./22.png)

Reviewer: ```suggestion
 ![Screenshot of Azure Arc-enabled SQL Server resource](./18.png)

- Select "Databases" under "Data Management". Click on "Change License Type".

 ![Screenshot of Azure Arc-enabled SQL Server resource](./19.png)

 ![Screenshot of Change License Type](./20.png)

- Choose "License with Software Assurance".

 ![Screenshot of software assurance](./21.png)

- The databases will be discovered and shown in the Azure Portal after a few seconds.

 ![Screenshot of Databases in Azure portal](./22.png)
```

## Clean up environment

> Complete the following steps to clean up your environment.
> 
> Remove the Azure Resource group from the portal.

Reviewer: ```suggestion
To clean up your deployment, simply delete the resource group from the portal.
```
